fuel/converters/adult.py

```diff
@@ -0,0 +1,140 @@
+
+    train_content = open(train_path, 'r').readlines()
+    test_content = open(test_path, 'r').readlines()
+    train_content = train_content[:-1]
+    test_content = test_content[1:-1]
+
+    features_list = []
+    targets_list = []
+    for content in [train_content, test_content]:
+        # strip out examples with missing features
+        content = [line for line in content if line.find('?') == -1]
+        # strip off endlines, separate entries
+        content = map(lambda l: l[:-1].split(', '), content)
+
+        features = map(lambda l: l[:-1], content)
+        targets = map(lambda l: l[-1], content)
+        del content
+        y = map(lambda l: [l[0] == '>'], targets)
+        y = numpy.array(y)
+        del targets
+
+        # Process features into a matrix
+        variables = [
+            'age', 'workclass', 'fnlwgt', 'education', 'education-num',
+            'marital-status', 'occupation', 'relationship', 'race', 'sex',
+            'capital-gain', 'capital-loss', 'hours-per-week', 'native-country'
+        ]
+        continuous = set([
+            'age', 'fnlwgt', 'education-num', 'capital-gain', 'capital-loss',
+            'hours-per-week'
+        ])
+
+        pieces = []
+        for i, var in enumerate(variables):
+            data = map(lambda l: l[i], features)
+            if var in continuous:
+                data = map(lambda l: float(l), data)
+                data = numpy.array(data)
+                data = data.reshape(data.shape[0], 1)
+            else:
+                unique_values = list(set(data))
+                data = map(lambda l: unique_values.index(l), data)
+                data = convert_to_one_hot(data)
+            pieces.append(data)
+
+        X = numpy.concatenate(pieces, axis=1)
+
+        features_list.append(X)
+        targets_list.append(y)
+
+    # the largets value in the last variable of test set is only 40, thus
+    # the one hot representation has 40 at the second dimention. While in
+    # training set it is 41. Since it lies in the last variable, so it is
+    # safe to simply add a last column with zeros.
+    features_list[1] = numpy.concatenate(
+        (features_list[1],
+         numpy.zeros((features_list[1].shape[0], 1),
+                     dtype=features_list[1].dtype)),
+        axis=1)
+    h5file = h5py.File(output_path, mode='w')
+    data = (('train', 'features', features_list[0]),
+            ('train', 'targets', targets_list[0]),
+            ('test', 'features', features_list[1]),
+            ('test', 'targets', targets_list[1]))
+
+    fill_hdf5_file(h5file, data)
+    h5file['features'].dims[0].label = 'batch'
+    h5file['features'].dims[1].label = 'feature'
+    h5file['targets'].dims[0].label = 'batch'
+    h5file['targets'].dims[1].label = 'index'
+
+    h5file.flush()
+    h5file.close()
+
+    return (output_path,)
+
+
+def fill_subparser(subparser):
+    subparser.set_defaults(func=convert_adult)
```

Comment:
The way `fill_subparser` is implemented has changed slightly. You now need to return the conversion function instead of setting it as `func`'s default value, i.e.

``` python
def fill_subparser(subparser):
    return convert_adult
```

I forgot to mention it, sorry about that!